planner/core/find_best_task.go

```diff
@@ -290,6 +290,12 @@ func (p *baseLogicalPlan) enumeratePhysicalPlans4Task(physicalPlans []PhysicalPl
 
 // compareTaskCost compares cost of curTask and bestTask and returns whether curTask's cost is smaller than bestTask's.
 func compareTaskCost(ctx sessionctx.Context, curTask, bestTask task) (curIsBetter bool, err error) {
+	if curTask.invalid() {
```

Comment:
In some cases, `curTask` can be invalid with 0 cost like `*mppTask{p=nil, cst=0}`, and then it can pass the condition `curCost < bestCost || ...` and set the `bestTask` to invalid.
To prevent the case above, we have to check whether `curTask` is valid first.